src/py/flwr/common/message.py

```diff
@@ -221,31 +240,147 @@ def __eq__(self, other: object) -> bool:
 
 
 class Message:
-    """State of your application from the viewpoint of the entity using it.
+    """Represents a message exchanged between ClientApp and ServerApp.
+
+    This class encapsulates the payload and metadata necessary for communication
+    between a ClientApp and a ServerApp.
 
     Parameters
     ----------
-    metadata : Metadata
-        A dataclass including information about the message to be executed.
-    content : Optional[RecordSet]
+    content : Optional[RecordSet] (default: None)
         Holds records either sent by another entity (e.g. sent by the server-side
         logic to a client, or vice-versa) or that will be sent to it.
-    error : Optional[Error]
+    error : Optional[Error] (default: None)
         A dataclass that captures information about an error that took place
         when processing another message.
+    dst_node_id : Optional[int] (default: None)
+        An identifier for the node receiving this message.
+    message_type : Optional[str] (default: None)
+        A string that encodes the action to be executed on
+        the receiving end.
+    ttl : Optional[float] (default: None)
+        Time-to-live (TTL) for this message in seconds. If `None` (default),
+        the TTL is set to 43,200 seconds (12 hours).
+    group_id : Optional[str] (default: None)
+        An identifier for grouping messages. In some settings, this is used as
+        the FL round.
+    reply_to : Optional[Message] (default: None)
+        The message to which this message is a reply. This message does not retain
+        the original message's content but derives its metadata from it.
     """
 
-    def __init__(
+    @overload
+    def __init__(  # pylint: disable=too-many-arguments  # noqa: E704
         self,
-        metadata: Metadata,
+        content: RecordSet,
+        dst_node_id: int,
+        message_type: str,
+        *,
+        ttl: float | None = None,
+        group_id: str | None = None,
+    ) -> None: ...
+
+    @overload
+    def __init__(  # noqa: E704
+        self, content: RecordSet, *, reply_to: Message, ttl: float | None = None
+    ) -> None: ...
+
+    @overload
+    def __init__(  # noqa: E704
+        self, error: Error, *, reply_to: Message, ttl: float | None = None
```

Comment:
```suggestion
        self, error: Error | None = None, *, reply_to: Message, ttl: float | None = None
```

Should `Error` default to `None`?

Reply:
Resolving based on my earlier comment.

Reply:
No, the `error` is required. This is one of the overload signature.